md2notion/upload.py

```diff
@@ -115,34 +118,49 @@ def filesFromPathsUrls(paths):
             fileLike.name = path
             yield (path, fileName, fileLike)
         else:
-            for path in glob.glob(path, recursive=True):
+            globPaths = glob.glob(path, recursive=True)
+            if not globPaths:
+                raise RuntimeError(f'No file found for glob {path}')
+            for path in globPaths:
                 with open(path, "r", encoding="utf-8") as file:
                     yield (path, os.path.basename(path), file)
 
-if __name__ == "__main__":
-    import argparse
-    import sys
-    from notion.block import PageBlock
-    from notion.client import NotionClient
-
+def cli(argv):
     parser = argparse.ArgumentParser(description='Uploads Markdown files to Notion.so')
     parser.add_argument('token_v2', type=str,
                         help='the token for your Notion.so session')
     parser.add_argument('page_url', type=str,
                         help='the url of the Notion.so page you want to upload your Markdown files to')
     parser.add_argument('md_path_url', type=str, nargs='+',
                         help='A path, glob, or url to the Markdown file you want to upload')
+    parser.add_argument('--create', action='store_const', dest='mode', const='create',
+                        help='Create a new child page (default)')
+    parser.add_argument('--append', action='store_const', dest='mode', const='append',
+                        help='Append to page instead of creating a child page')
+    parser.add_argument('--clear-previous', action='store_const', dest='mode', const='clear',
+                        help='Clear a previous child page with the same name if it exists')
+    parser.set_defaults(mode='create')
 
-    args = parser.parse_args(sys.argv[1:])
+    args = parser.parse_args(argv)
 
     print("Initializing Notion.so client...")
     client = NotionClient(token_v2=args.token_v2)
     print("Getting target PageBlock...")
     page = client.get_block(args.page_url)
+    uploadPage = page
 
     for mdPath, mdFileName, mdFile in filesFromPathsUrls(args.md_path_url):
-        # Make the new page in Notion.so
-        pageName = mdFileName[:40]
```

Comment:
Curious, was there any good reason for truncating the filename to 40 characters before? (A limitation of notion?)

Reply:
Nah, just being a little cautious with the API because notion-py doesn't do that many security checks IIRC. I noticed a really long name I had added worked from my blog stuff so I think it should be fine.